### PR TITLE
[makefile] add macro OPENTHREAD_CONFIG_MAC_SSED_TO_SSED_LINK_ENABLE

### PR DESCRIPTION
**Pull request by brianljt**

### CODE REVIEW

File: src/core/config/mac.h [hunk 1]
```diff
@@ -407,4 +407,14 @@
 #define OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD 2
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_MAC_SSED_TO_SSED_LINK_ENABLE
+ *
+ * This setting configures the CSL SSED-to-SSED link feature in Thread 1.2.
```
Comment (Irving-cl) [inline src/core/config/mac.h:413]:
```suggestion
 * SSED-to-SSED link feature option. This is not an official Thread 1.2 feature.
```

**Review by Irving-cl — APPROVED** [review]
LGTM

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

File: etc/cmake/options.cmake [hunk 2]
```diff
@@ -132,6 +132,11 @@ if(OT_CSL_DEBUG)
     target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_MAC_CSL_DEBUG_ENABLE=1")
 endif()
 
+option(OT_SSED_SSED_LINK "enable SSED-to-SSED link")
```
Comment (simonlingoogle) [inline etc/cmake/options.cmake:135]:
```suggestion
option(OT_SSED_LINK "enable SSED-to-SSED link")
```

`SSED_LINK` is clear enough?

Comment (brianljt) [reply etc/cmake/options.cmake:135]: `SSED_LINK` could mean SSED to a Thread Router. From Spec, it is called SSED-TO-SSED link.

File: examples/README.md [hunk 3]
```diff
@@ -56,6 +56,7 @@ This page lists the available common switches with description. Unless stated ot
 | SLAAC | OT_SLAAC | Enables support for adding auto-configured SLAAC addresses by OpenThread. This feature is enabled by default. |
 | SNTP_CLIENT | OT_SNTP_CLIENT | Enables support for SNTP Client. |
 | SPINEL_ENCRYPTER_LIBS | not implemented | Specifies library files (absolute paths) for implementing the NCP Spinel Encrypter. |
+| SSED_SSED_LINK | OT_SSED_SSED_LINK | Enables the optional Thread 1.2 feature of SSED-to-SSED Link. |
```
Comment (Irving-cl) [inline examples/README.md:59]: I'm thinking if we may add the doc after we submit the whole feature. Users may get confused: it's in the doc but cannot be used.

Comment (brianljt) [reply examples/README.md:59]: Agree. We can add it to README after the feature is completed.

**Comment by brianljt** [comment]
Implement this feature in a separate branch first. Close this PR.

**Pull request closed without merging**